ssz/utils.py

```diff
@@ -143,28 +142,21 @@ def hash_layer(child_layer: Sequence[bytes]) -> Tuple[Hash32, ...]:
     return parent_layer
 
 
-def merkleize(chunks: Sequence[Hash32],
-              limit: int=None,
-              cache: CacheObj=None) -> Union[Hash32, Tuple[Hash32, CacheObj]]:
-    with_cache = cache is not None
-
-    chunk_len = len(chunks)
-
-    if limit is None:
-        limit = chunk_len
-
-    if limit == 0:
-        if cache is None:
-            return ZERO_HASHES[0]
-        else:
-            return ZERO_HASHES[0], cache
-
+def _get_lengths(chunks: Sequence[Hash32], limit: int, chunk_len: int) -> Tuple[int, int]:
     chunk_depth = max(chunk_len - 1, 0).bit_length()
     max_depth = max(chunk_depth, (limit - 1).bit_length())
     if max_depth > len(ZERO_HASHES):
         raise ValueError(f"The number of layers is greater than {len(ZERO_HASHES)}")
 
-    tmp_list = [None for _ in range(max_depth + 1)]
+    return chunk_depth, max_depth
+
+
+def _get_temp_merklized_result(chunks: Sequence[Hash32],
```

Comment:
Can we find a better name for this function or at least add a comment? It's not clear to me what the `temp_list` is.